examples/editor/src/App.tsx

```diff
@@ -1,26 +1,150 @@
 // import logo from './logo.svg'
 import "@blocknote/core/style.css";
-import { BlockNoteView, useBlockNote } from "@blocknote/react";
+import {
+  BlockNoteView,
+  getDefaultReactSlashMenuItems,
+  useBlockNote,
+  Theme,
+  defaultTheme,
+} from "@blocknote/react";
 import styles from "./App.module.css";
+import { BlockSchema, defaultBlockSchema } from "@blocknote/core";
+import { Image, insertImage } from "../../../tests/utils/customblocks/Image";
+import {
+  insertReactImage,
+  ReactImage,
+} from "../../../tests/utils/customblocks/ReactImage";
 
 type WindowWithProseMirror = Window & typeof globalThis & { ProseMirror: any };
 
+const redTheme: Partial<Theme> = {
+  colors: {
+    editor: {
+      text: {
+        light: "#222222",
+        dark: "#ffffff",
+      },
+      background: {
+        light: "#ffffff",
+        dark: "#9b0000",
+      },
+    },
+    menu: {
+      text: {
+        light: "#ffffff",
+        dark: "#ffffff",
+      },
+      background: {
+        light: "#9b0000",
+        dark: "#9b0000",
+      },
+    },
+    tooltip: {
+      text: {
+        light: "#ffffff",
+        dark: "#ffffff",
+      },
+      background: {
+        light: "#b00000",
+        dark: "#b00000",
+      },
+    },
+    hovered: {
+      text: {
+        light: "#ffffff",
+        dark: "#ffffff",
+      },
+      background: {
+        light: "#b00000",
+        dark: "#b00000",
+      },
+    },
+    selected: {
+      text: {
+        light: "#ffffff",
+        dark: "#ffffff",
+      },
+      background: {
+        light: "#c50000",
+        dark: "#c50000",
+      },
+    },
+    disabled: {
+      text: {
+        light: "#9b0000",
+        dark: "#9b0000",
+      },
+      background: {
+        light: "#7d0000",
+        dark: "#7d0000",
+      },
+    },
+    shadow: {
+      light: "#640000",
+      dark: "#640000",
+    },
+    border: {
+      light: "#870000",
+      dark: "#870000",
+    },
+    sideMenu: {
+      light: "#bababa",
+      dark: "#ffffff",
+    },
+    // TODO: Update
+    highlightColors: defaultTheme.colors.highlightColors,
+  },
+  borderRadius: 4,
+  fontFamily: "Helvetica Neue, sans-serif",
+};
+
+const schema = {
+  ...defaultBlockSchema,
+  image: Image,
+  reactImage: ReactImage,
+} satisfies BlockSchema;
+
 function App() {
   const editor = useBlockNote({
     onEditorContentChange: (editor) => {
       console.log(editor.topLevelBlocks);
     },
-    editorDOMAttributes: {
-      class: styles.editor,
-      "data-test": "editor",
+    blockSchema: schema,
+    slashMenuItems: [
+      ...getDefaultReactSlashMenuItems(schema),
+      insertImage,
+      insertReactImage,
+    ],
+    domAttributes: {
```

Comment:
should we be able to pass custom domattributes for blocks here?

Reply:
Don't see why not, I think most people will still just add classes but imo it adds very little complexity so I think we may as well give people the option.